container/container_unix.go

```diff
@@ -280,18 +281,50 @@ func (container *Container) UnmountSecrets() error {
 
 // UpdateContainer updates configuration of a container.
 func (container *Container) UpdateContainer(hostConfig *containertypes.HostConfig) error {
+	sysInfo := sysinfo.New(true)
+
 	container.Lock()
 	defer container.Unlock()
 
 	// update resources of container
 	resources := hostConfig.Resources
 	cResources := &container.HostConfig.Resources
+
+	// validate NanoCPUs, CPUPeriod, and CPUQuota
+	if resources.NanoCPUs > 0 && resources.CPUPeriod > 0 {
+		return fmt.Errorf("Conflicting options: Nano CPUs and CPU Period cannot both be updated")
+	}
+	if resources.NanoCPUs > 0 && resources.CPUQuota > 0 {
+		return fmt.Errorf("Conflicting options: Nano CPUs and CPU Quota cannot both be updated")
+	}
+	if resources.NanoCPUs > 0 && (!sysInfo.CPUCfsPeriod || !sysInfo.CPUCfsQuota) {
+		return fmt.Errorf("NanoCPUs can not be updated, as your kernel does not support CPU cfs period/quota or the cgroup is not mounted")
+	}
+	if resources.NanoCPUs < 0 || resources.NanoCPUs > int64(sysinfo.NumCPU())*1e9 {
+		return fmt.Errorf("Range of Nano CPUs is from 1 to %d", int64(sysinfo.NumCPU())*1e9)
+	}
+	if resources.NanoCPUs > 0 && cResources.CPUPeriod > 0 {
```

Comment:
For example, this validation applies both to _create_ and _update_ of a container; they are never allowed to both be set (https://github.com/docker/docker/blob/f7819fcb25aaf5b08202db275847c825e7b09999/daemon/daemon_unix.go#L369-L371)